### PR TITLE
Fix golint errors in prow packages.

### PR DESCRIPTION
**Pull request by ibrasho**
Continuation of #9254.

I just rebased on latest master and I'll work on fixing the issues before completing the work.

Resolves: #9247

### CODE REVIEW

File: label_sync/labels.yaml [hunk 1]
```diff
@@ -715,8 +715,8 @@ repos:
         target: both
         addedBy: label
       - color: 0052cc
-        description: Issues or PRs related to prow's artifact-uploader component
-        name: area/prow/artifact-uploader
```
Comment (ibrasho) [inline label_sync/labels.yaml]: Should we rename the label?

Comment (fejta) [reply label_sync/labels.yaml]: yes, please migrate the old label to the new one

File: prow/apis/prowjobs/v1/register.go [hunk 2]
```diff
@@ -37,12 +37,13 @@ func Resource(resource string) schema.GroupResource {
 	return SchemeGroupVersion.WithResource(resource).GroupResource()
 }
 
+// SchemeBuilder : TODO: (alisondy) Document what SchemeBuilder is used for
```
Comment (fejta) [inline prow/apis/prowjobs/v1/register.go]: Please remove all comments like this. Either add the documentation or leave it undocumented.

This silences the lint error without actually addressing the intent of the lint error.

File: prow/plugins/approve/approvers/owners.go [hunk 3]
```diff
@@ -169,7 +173,7 @@ func (o Owners) GetOwnersSet() sets.String {
 	return owners
 }
 
-// Shuffles the potential approvers so that we don't always suggest the same people
+// GetShuffledApprovers : Shuffles the potential approvers so that we don't always suggest the same people
```
Comment (fejta) [inline prow/plugins/approve/approvers/owners.go]: The format is `// method shuffles blah blah blah`

AKA
* The first word of the sentence is the method name. 
* There's no colon
* the second word of the sentence is lowercase.

**Comment by ibrasho** [comment]
I just went through the old comments and discovered that. I've reset to HEAD and will start working on them package by package.

**Comment by ibrasho** [comment]
@fejta I feel the amount of change in this PR is enough for a review.

The remaining issues are in `prow/plugins` and I can submit them in another PR if that's okay.

File: label_sync/labels.yaml (hunk 1 above)
Comment (fejta) [reply label_sync/labels.yaml]: Use the `previously` field to migrate the old value to the new one: https://github.com/kubernetes/test-infra/tree/master/label_sync#configuration

File: prow/cmd/artifactuploader/OWNERS [hunk 4]
```diff
@@ -1,4 +1,4 @@
 approvers:
 - stevekuznetsov
 labels:
- - area/prow/artifact-uploader
+ - area/prow/artifactuploader
```
Comment (fejta) [inline prow/cmd/artifactuploader/OWNERS]: Nit: can you have the `- area` align with `labels:`

File: prow/cmd/artifactuploader/BUILD.bazel [hunk 5]
```diff
@@ -26,7 +26,7 @@ prow_image(
 )
 
 go_binary(
-    name = "artifact-uploader",
+    name = "artifactuploader",
```
Comment (fejta) [inline prow/cmd/artifactuploader/BUILD.bazel]: Can we move the artifactuploader rename into its own PR (and asssign stevekuznetsov)?

File: prow/config/secrets_agent.go [hunk 6]
```diff
@@ -14,9 +14,10 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-// Implements an agent to read and reload the secrets.
 package config
 
+// Implements an agent to read and reload the secrets.
```
Comment (fejta) [inline prow/config/secrets_agent.go]: This file should probably be in its own package..

Comment (ibrasho) [reply prow/config/secrets_agent.go]: Under `prow/config` or else where?

Comment (ibrasho) [reply prow/config/secrets_agent.go]: Not sure if I got you right, but here is an attampt: https://github.com/kubernetes/test-infra/commit/c7f7e9fc19297e78a4dd38fbf205e71f2068326c

Comment (fejta) [reply prow/config/secrets_agent.go]: Thanks! That looks like a great PR in its own right!

**Comment by fejta** [comment]
/lgtm
/hold

/assign @cjwagner @stevekuznetsov 

This PR changes a bunch of exported field names. They all seem perfectly sensible to me (ID instead of Id, config.Delta instead of config.ConfigDelta, etc) but are technically breaking. Holler if you want to argue against it. Otherwise Ibrahim feel free to `/hold cancel` Monday evening PST.

> make it another PR?

Yes, in general the smaller the PR the easier it is to review.

**Comment by stevekuznetsov** [comment]
/hold

Can we not rename `artifact-uploader`? That's going to have reverberating changes downstream for anyone actually using it and I'm not sure it's worth it.

**Comment by fejta** [comment]
@stevekuznetsov this PR doesn't rename `artifact-uploader`. I agree that is a most disruptive change and asked we move it into https://github.com/kubernetes/test-infra/pull/10444

The changes here are:
* `config.ConfigDelta` => `config.Delta` (agent.go, L96)
*  `v1.SchemeBuilder` => `v1.schemeBuilder` (regsiter.go, L41)
* `jobs.ErrProwJobNotFound` => `jobs.errProwJobNotFound` (jobs.go, L43)
* `github.Hook.Id` => `github.Hook.ID` (types.go L345)
* `githuboauth.NewGithubOAuthAgent` => `github.NewAgent` (githuboauth.go, L84)
* `githuboauth.GithubOAuthAgent` => `github.Agent` (githuboauth.go, L78)
* `prstatus.PullRequestQueryHandler.HeadContexts` => `prstatus.PullRequestQueryHandler.GetHeadContexts` (prstatus.go, L55)
* `prstatus.PullRequestWithContext` => `prstatus.PullRequestWithContexts` (prstatus.go, L68)

I think all of these changes make sense

File: prow/git/git.go [hunk 7]
```diff
@@ -244,14 +244,17 @@ func (r *Repo) CheckoutNewBranch(branch string) error {
...
-	if b, err := co.CombinedOutput(); err == nil {
+
+	b, err := co.CombinedOutput()
+	if err == nil {
 		return true, nil
-	} else {
-		r.logger.WithError(err).Warningf("Merge failed with output: %s", string(b))
 	}
+
+	r.logger.WithError(err).Warningf("Merge failed with output: %s", string(b))
```
Comment (stevekuznetsov) [inline prow/git/git.go]: I would group this with the above stanza, not the lower one

File: prow/hook/plugins.go [hunk 8]
```diff
@@ -19,7 +19,7 @@ package hook
 // All enabled plugins. We need to empty import them like this so that they
 // will be linked into any hook binary.
 import (
-	_ "k8s.io/test-infra/prow/plugins/approve"
+	_ "k8s.io/test-infra/prow/plugins/approve" // Ensure all packages are loaded.
```
Comment (stevekuznetsov) [inline prow/hook/plugins.go]: Not sure about the utility of this change given the comment above the import block

File: prow/initupload/doc.go [hunk 9]
```diff
@@ -14,7 +14,6 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-// Package initupload determines the output status of clone
-// operations and posts that status along with artifacts and
-// logs to cloud storage
+// Package initupload determines the output status of clone operations and posts that status along
```
Comment (stevekuznetsov) [inline prow/initupload/doc.go]: Does `golint` care about comment line length now? Yuck.

**Comment by fejta** [comment]
Yes, planning to add a golint check once we are golint clean :)

**Comment by fejta** [comment]
Can you address the three oustanding comments from Steve? After that I think this is good to go!

File: prow/hook/plugins.go (hunk 8 above)
Comment (ibrasho) [reply prow/hook/plugins.go]: `golint` expects the comment to be on the same line as the first `_` aliased import. Should I remove the one before the `import` statement?

**Comment by ibrasho** [comment]
Just pushed some changes that should resolve of Steve's comments.

**Comment by ibrasho** [comment]
Some upstream changes used `config.ConfigDelta`, so I had to rebase on master again and update the new code.
